spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/aot/SpannerRuntimeHints.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.data.spanner.aot;
+
+import java.util.Arrays;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+
+public class SpannerRuntimeHints implements RuntimeHintsRegistrar {
+
+  @Override
+  public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+    hints
+        .reflection()
+        .registerTypes(
+            Arrays.asList(
+                TypeReference.of(
+                    com.google.cloud.spring.data.spanner.repository.support.SimpleSpannerRepository
+                        .class),
+                TypeReference.of(java.lang.String.class)),
```

Comment:
Needing to include `String` here seems odd. Do we know why this is needed?

Reply:
Included a comment with some context but looking at the stacktrace, this type is referenced from our code`SpannerPersistentEntityImpl.tableName()` which then leads to a reflective call down the line within the Spring code:

 https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/c92260f1b583575436a61ec8f6fb2252737cd43f/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerPersistentEntityImpl.java#L351-L360:
```
com.oracle.svm.core.jdk.UnsupportedFeatureError: Runtime reflection is not supported for public java.lang.String java.lang.String.concat(java.lang.String)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:89)
	at java.base@17.0.7/java.lang.reflect.Method.acquireMethodAccessor(Method.java:76)
	at java.base@17.0.7/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute(ReflectiveMethodExecutor.java:139)
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:112)
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:380)
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:96)
	at org.springframework.expression.spel.ast.SpelNodeImpl.getTypedValue(SpelNodeImpl.java:119)
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:309)
	at com.google.cloud.spring.data.spanner.core.mapping.SpannerPersistentEntityImpl.tableName(SpannerPersistentEntityImpl.java:361)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlString(SpannerSchemaUtils.java:85)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.lambda$getDropTableDdlStringsForInterleavedHierarchy$1(SpannerSchemaUtils.java:280)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDdlStringForInterleavedHierarchy(SpannerSchemaUtils.java:297)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlStringsForInterleavedHierarchy(SpannerSchemaUtils.java:275)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlStringsForInterleavedHierarchy(SpannerSchemaUtils.java:160)

```